src/main/scala/net/elodina/mesos/dse/HttpServer.scala

```
@@ -384,9 +395,18 @@ object HttpServer {
 
       for (node <- nodes) {
         // check node is running, starting, stopping, because it's state could have changed
-        checkState(node)
+        try {
+          checkState(node)
+        } catch {
+          case e: HttpError if withProgress && node != nodes(0) =>
```

Strange catch clause. May be just let exception be thrown ?